gidgethub/sansio.py

```diff
@@ -308,8 +308,8 @@ def decipher_response(status_code: int, headers: Mapping[str, str],
             elif status_code == 422:
                 errors = data.get("errors", None)
                 if errors:
-                    fields = ", ".join(repr(e["field"]) for e in errors)
-                    message = f"{message} for {fields}"
+                    error_context = ", ".join(repr(e.get("field", e["resource"])) for e in errors)
```

Comment:
The resource isn't important, it's the reason for the failure that is (hence why we weren't pulling in the `"resource"` value even when it was available.

It would be better to have logic that is specific to each case where if `"field"` exists the current semantics occur, and if `"code"` is `"custom"` then return something like `f"{message}: {e['message']}"`. You will also need a new exception like `ValidationError` which `InvalidField` inherits from.

Reply:
What I was curious about was the case where there were some errors that had `"code": "custom"`, and some that had a `"field"`label. Maybe I'm just trying to pre-optimize this, because I don't know if that even can occur. I'll go ahead and rework this with the assumption that all errors look the same, but that might be a case that needs to be handled. 

In the case of mixed error types, I don't know which exception to raise or how to format the message (since both have a different format). Any thoughts/suggestions on this?